src/server_manager/model/server.ts

```diff
@@ -92,6 +92,8 @@ export interface ManualServer extends Server {
 // Managed servers are servers created by the Outline Manager through our
 // "magic" user experience, e.g. DigitalOcean.
 export interface ManagedServer extends Server {
+  // Returns the unique cloud account ID.
+  getAccountId(): string;
```

Comment:
As I think more about this, I'm not completely convinced a inner object should have to know about the outer object. And it should have to know how to return a globally unique identifier. It's kind of a circular dependency. It's up to the caller to create the globally unique identifier.

When we are loading the servers, we know exactly what account we are getting it from, so why do we need this id on the server?

I think the global id logic can actually live in the app code instead. As you load them and create the UI objects, create the global ids.

Reply:
It would be helpful to see how the ids will actually be used. I suggest creating a PR on top of this that uses the ids to remove all the isManagedServer == isDigitalOcean usage and for event handling.

Reply:
Interesting observation.

We do know which account we're getting the server from so we do it in the app.ts/UI logic. This requires a fair bit of plumbing to assign the correct account ID (or no account ID for manual servers?).

I'm exploring this in a separate PR.